Formula/p/percona-toolkit.rb

```diff
@@ -27,13 +28,19 @@ class PerconaToolkit < Formula
 
   depends_on "go" => :build
   depends_on "mysql-client"
-  depends_on "openssl@3"
 
   uses_from_macos "perl"
-  uses_from_macos "zlib", since: :sonoma
 
-  on_intel do
-    depends_on "zstd"
+  on_macos do
+    depends_on "openssl@3"
+
+    on_intel do
+      depends_on "zstd"
+    end
+  end
+
+  on_sonoma :or_newer do
+    depends_on "zlib"
   end
```

Comment:
May be worth a follow up PR to see if `-dead_strip_dylibs` can help align this odd dependency tree.

Reply:
#179760